api/src/auth/drivers/oauth2.ts

```diff
@@ -162,8 +156,10 @@ export class OAuth2AuthDriver extends LocalAuthDriver {
 				? payload['codeVerifier']
 				: generators.codeChallenge(payload['codeVerifier']);
 
+			const callbackUrl = getCallbackFromOriginUrl(this.redirectUris, payload['originUrl']);
+
 			tokenSet = await this.client.oauthCallback(
-				this.redirectUrl,
+				callbackUrl?.toString(),
```

Comment:
Should this be possibly undefined when previously it was not?

Reply:
Update this part